### PR TITLE
DBZ-6213 Adjust connector restart

### PR DESCRIPTION
**Pull request by vjuranek**
* Try to start the connector in BaseSourceTask::start(). This is the root cause of DBZ-6213 as EmbeddedEngine calls start() and assumes it really tries to start the connector.
* Simplify the code by removing unnecessary checks and restartDelay.hasElapsed() call.

https://issues.redhat.com/browse/DBZ-6213

### CODE REVIEW

File: debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java
Comment (jeremy-l-ford) [inline debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java:146]: Could just catch the RetriableException and avoid the instanceof check.

**Review by jeremy-l-ford — APPROVED** [review]

File: debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java
Comment (vjuranek) [reply debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java:146]: right, thanks for noting it! Fixed.

**Review by jeremy-l-ford — APPROVED** [review]

File: debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java
Comment (jpechane) [inline debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java:260]: Aren't you converting `Duration` to `Duration` here?

File: debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java
Comment (jpechane) [inline debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java:315]: IMHO by removing it you ae removing the pause.

File: debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java
Comment (jpechane) [inline debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java:147]: I think there is a method that takes Duration as parameter.

**Review by jpechane — CHANGES_REQUESTED** [review]
@vjuranek LGTM. I left few minor comments.

File: debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java
Comment (vjuranek) [reply debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java]: right, fixed

File: debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java
Comment (vjuranek) [reply debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java]: ops, yes:-( Fixed.

File: debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java
Comment (vjuranek) [reply debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java:315]: Right, but only the first one pause:-) [This method](https://github.com/debezium/debezium/blob/main/debezium-core/src/main/java/io/debezium/util/ElapsedTimeStrategy.java#L50) says, if the delay interval has already elapsed or not. As a side effect it recomputes next timestamp for waiting. This method is very confusing (it's description, side effects and initialization) and IMHO should be also refactored.

However, you are right that removing it is wrong - due to its side effect if this call is removed, the elapsed interval won't be initialized and actually won't do any wait when the strategy is used for the first time. So basically this has to be called every time right after `ElapsedTimeStrategy` is created.

**Review by vjuranek — COMMENTED** [review]
@jpechane all the issues should be fixed now, thanks for spotting them, especially the one with removing `restartDelay.hasElapsed()`.

**Review by jpechane — APPROVED** [review]
Waiting for CI to complete before merging.

File: debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java
Comment (jpechane) [reply debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java:315]: @vjuranek I agree on the refactoring

**Comment by jpechane** [comment]
@vjuranek Applied, thanks a lot

**Pull request merged**
